### PR TITLE
Strict mode compliance

### PR DESCRIPTION
**Pull request by johnspackman**
Various minor fixes for strict mode compliance, IE remove references to `arguments.callee` and `arguments.caller`

### CODE REVIEW

File: framework/source/class/qx/test/Bootstrap.js [hunk 1]
```diff
@@ -292,7 +292,7 @@ qx.Class.define("qx.test.Bootstrap",
 
           stopEngine : function()
           {
-            var ret = arguments.callee.base.call();
+            var ret = qx.test.Bmw.prototype.stopEngine.base.call();
```
Comment (derrell) [inline framework/source/class/qx/test/Bootstrap.js]: Is this the same as the following?
```
var ret = this.base(arguments);
```

.call() with no first argument means that the context is explicitly excluded. That sounds wrong to start with, unless there is no reference to `this` in the `...stopEngine.base` method...

Comment (derrell) [inline framework/source/class/qx/test/Bootstrap.js]: Similarly, elsewhere in here. This is *really* old code you're modifying here.

**Review by derrell — CHANGES_REQUESTED** [review]

File: framework/source/class/qx/test/Bootstrap.js (hunk 1 above)
Comment (johnspackman) [inline framework/source/class/qx/test/Bootstrap.js]: yes, really old - I guess this is why it was doing some slightly strange things to accomplish the tests; I was aiming to make the smallest changes and just translate the syntax, but I agree it doesn't make it any clearer!  I'll change it

File: framework/source/class/qx/util/Function.js [hunk 2]
```diff
@@ -50,54 +50,44 @@ qx.Bootstrap.define("qx.util.Function", {
      * @return {Function} a wrapper function which <em>shields</em> the given callback function
      */
     debounce: function (callback, delay, immediate) {
+      var fired = false;
+      var intervalId = null;
       var wrapperFunction = function () {
-        arguments.callee.immediate = !!(immediate);
-
         // store the current arguments at the function object
         // to have access inside the interval method
```
Comment (derrell) [inline framework/source/class/qx/util/Function.js]: This comment is now incorrect

**Review by derrell — CHANGES_REQUESTED** [review]

**Review by cboulanger — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Review by cajus — APPROVED** [review]

**Pull request merged**